### DISCUSSION TITLE
Issue Running on Ubuntu with Docker from Snap

### DISCUSSION THREAD

**Discussion by smozely** [batect/batect#880] [Q&A]
Hi there, I've been using batect on a project and it works great ... however on an local Ubuntu Machine I'm getting this error whenever I try to run a batect task ... 

**Error:**
```
Error: Could not run container app.
Starting container 'd963231ff6ae1d0296ddd37d262e9271cbca02f57909d6224168b619d331569c' failed: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/tmp/batect-passwd-8875828100056150442\\\" to rootfs \\\"/var/snap/docker/common/var-lib-docker/overlay2/d1014d45af18268b0dfd969e40d8d4366e11d4f2e16d9ff552c589277b4e694a/merged\\\" at \\\"/var/snap/docker/common/var-lib-docker/overlay2/d1014d45af18268b0dfd969e40d8d4366e11d4f2e16d9ff552c589277b4e694a/merged/etc/passwd\\\" caused \\\"not a directory\\\"\"": unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type
```

**Running these versions:**
Ubuntu: 20.04
Docker: 19.03.13
Batect: 0.70.1

Just wondering if this has been reported before or workaround is known?

**Comment by charleskorn** [comment]
Hi @smozely, Ubuntu is definitely supported. 

Could you tell me a little more about how Docker was installed on your machine? Was it through Docker's official apt repository, or have you used another mechanism? It looks like Docker is set up in a different way to what I have, and I can't reproduce the problem. 

Can you also try disabling `run_as_current_user` and see if that fixes the issue?

**Reply by smozely** [reply]
Ah sweet, hadn't thought about how I installed docker ... solved it now ...

I had installed Docker using snap, which I remember being the first time I had ever used snap but some docs online seemed to be recommending it as the "new way" of installing things in Ubuntu (had been a long time since I'd last played with raw Linux distros, so just followed what I read).

... a google now seems there are a few people having problems with snap

So uninstalled it, and re-installed it the apt way, and everything is happy. Plus apt had a more recent version of Docker as well
